internal/provider/context_field_resource.go

```diff
@@ -0,0 +1,320 @@
+package provider
+
+import (
+	"context"
+	"fmt"
+
+	unleash "github.com/Unleash/unleash-server-api-go/client"
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/resource"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/planmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/resource/schema/stringplanmodifier"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"github.com/hashicorp/terraform-plugin-framework/types/basetypes"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
+)
+
+var (
+	_ resource.Resource                = &contextFieldResource{}
+	_ resource.ResourceWithConfigure   = &contextFieldResource{}
+	_ resource.ResourceWithImportState = &contextFieldResource{}
+)
+
+func NewContextFieldResource() resource.Resource {
+	return &contextFieldResource{}
+}
+
+type contextFieldResource struct {
+	client *unleash.APIClient
+}
+
+type contextFieldResourceModel struct {
+	Name        types.String        `tfsdk:"name"`
+	Description types.String        `tfsdk:"description"`
+	Stickiness  types.Bool          `tfsdk:"stickiness"`
+	LegalValues basetypes.ListValue `tfsdk:"legal_values"`
+}
+
+func (r *contextFieldResource) Configure(ctx context.Context, req resource.ConfigureRequest, _ *resource.ConfigureResponse) {
+	if req.ProviderData == nil {
+		return
+	}
+
+	client, ok := req.ProviderData.(*unleash.APIClient)
+	if !ok {
+		tflog.Error(ctx, "Unable to prepare client")
+		return
+	}
+	r.client = client
+}
+
+func (r *contextFieldResource) Metadata(_ context.Context, req resource.MetadataRequest, resp *resource.MetadataResponse) {
+	resp.TypeName = req.ProviderTypeName + "_context_field"
+}
+
+func (r *contextFieldResource) Schema(_ context.Context, _ resource.SchemaRequest, resp *resource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Description: "Fetch a context field.",
+		Attributes: map[string]schema.Attribute{
+			"name": schema.StringAttribute{
+				Description: "The name of the context field.",
+				Required:    true,
+				PlanModifiers: []planmodifier.String{
+					stringplanmodifier.RequiresReplace(), // name is effectively the id so changing this means making a few context field
+				},
+			},
+			"description": schema.StringAttribute{
+				Description: "A description of the context field.",
+				Optional:    true,
+				PlanModifiers: []planmodifier.String{
```

Comment:
Description is only ignored if it's null. This does mean that you can never unset a description without a recreate. Rather than argue with someone that an empty string is not the same as a null string, I've chosen to let this just recreate. Description shouldn't change that often so I think it's okay